keyboards/gmmk/numpad/readme.md

```diff
@@ -0,0 +1,29 @@
+# GMMK NUMPAD
+
+A 17 Key macropad made ans old by Glorious LLC, equiped with a WB32 microcontroller, with support for a rotary encoder.
+
+* Keyboard Maintainer: [GloriousThrall](https://github.com/GloriousThrall)
+* Hardware Supported: GMMK Numpad
+* Hardware Availability: [GloriousPCGaming.com](https://www.gloriousgaming.com/products/gmmk-numpad)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make gmmk/numpad:default
+
+Flashing example for this keyboard:
+
+    make gmmk/numpad:default:flash
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+## Bootloader
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the Num key and plug in the keyboard
+* **Keycode in layout**: Press the Num+Del key which is mapped to `QK_BOOT`
+
+## Broken
+
+* Slider does not function
+* Rotary input push button (mapped to gpio)
```

Comment:
If the push switch connects that GPIO to GND, you can add support for it by declaring that GPIO pin as an extra row pin (because your matrix is `ROW2COL`, and therefore row pins are inputs), then using any matrix location in that row for the encoder key (connecting a matrix input to GND will result in all matrix locations in that row appearing as pressed).  You can also use the `MATRIX_MASKED` feature to mask all other locations in that row; this will avoid some useless keymap lookups when the encoder is pressed.

If the push switch is connected in some other way, at the moment the only solution to support it properly is implementing a custom matrix (which would unfortunately duplicate a lot of the standard matrix code).

Reply:
I cannot find the pinout for the wb32f3g7. from using a continuity checker the rotary encoder sw and gnd are connected to 2 pins on the microprocessor

Reply:
they appear to be commented to the pins directly beside the placement marker (dot on the Microcontroller)

Reply:
Refer to the [docs](https://github.com/qmk/qmk_firmware/pull/16871#issuecomment-1436188572) here.
* https://github.com/qmk/qmk_firmware/pull/16871#issuecomment-1436188572

Reply:
@adophoxia is the pinout the same for the WB32F3G7 vs WB32FQ95?

Reply:
@sigprof I've discovered that it is connected to VDD and PB8, when I add PB8 to the matrix row I get the following warning 
```
⚠ gmmk/numpad: led_config: numpad.c: Unable to parse g_led_config matrix data
```

Reply:
Hub16 uses a custom matrix to get the encoder presses but this seems a little out of my scope of knowledge at the moment.

Reply:
I got the rotary button working however it wont be customizable with via atm

Reply:
You can configure the VIA Definition JSON that it'll be used for the encoder with this guide [`here`](https://www.caniusevia.com/docs/layouts#optional-rotary-encoder-combined-switchrotary-encoder-footprint).

Reply:
@adophoxia the issue is that the way I got it working was not adding it to the matrix. if you check the most recent commit you will understand.

Reply:
rotary is added just not the push button, sorry should have specified. the push button is fixed as KC_MUTE since its not part of the matrix